jujugui/static/gui/src/app/components/scale-service/scale-service.js

```diff
@@ -0,0 +1,164 @@
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('scale-service', function() {
+
+  juju.components.ScaleService = React.createClass({
+
+    getInitialState: function() {
+      return {
+        constraintsVisibility: false
+      };
+    },
+
+    /**
+      Event handler for the radio button selection which should show or
+      hide the constraint options.
+
+      @method _toggleConstraints
+      @param {Object} e The change event.
+    */
+    _toggleConstraints: function(e) {
+      var id = e.currentTarget.id;
+      this.setState({ constraintsVisibility: id === 'auto-place-units'});
+    },
+
+    /**
+      Generates the classes for each render for the constraints element.
+
+      @method _generateClasses
+    */
+    _generateClasses: function() {
+      return classNames(
+        'scale-service__constraints',
+        {
+          hidden: !this.state.constraintsVisibility
+        }
+      );
+    },
+
+    /**
+      When an input element value is changed by the user we update state
+      so that we don't have to query the DOM again when they submit.
+
+      @method _udpateState
+      @param {Object} e The change event.
+    */
+    _updateState: function(e) {
+      var currentTarget = e.currentTarget;
+      var state = {};
+      state[currentTarget.name] = currentTarget.value;
+      this.setState(state);
+    },
+
+    /**
+      Handles calling the appropriate methods to scale up the service.
+
+      @method _scaleUpService
+    */
+    _scaleUpService: function() {
+      var state = this.state;
+      var appState = {
+        sectionA: {
+          component: 'inspector',
+          metadata: {
+            id: this.props.serviceId,
+            activeComponent: 'units'
+          }}};
+      var numUnits = this.state['num-units'];
+      // Constraints will not be shown if the user wants to manually place.
+      if (!state.constraintsVisibility) {
+        // db, env, and service have already been bound to this function in
+        // the app.js definition.
+        this.props.addGhostAndEcsUnits(numUnits);
+        appState.sectionB = {
+          component: 'machine'
+        };
+      } else {
+        this.props.createMachinesPlaceUnits(numUnits, {
+          'cpu-power': this.state['cpu-constraint'],
+          'cpu-cores': this.state['cores-constraint'],
+          'mem': this.state['ram-constraint'],
+          'root-disk': this.state['disk-constraint']
+        });
+      }
+      this.props.changeState(appState);
+    },
+
+    render: function() {
+      var buttons = [{
+        title: 'Confirm',
+        action: this._scaleUpService
+      }];
+
+      return (
+        <div className="scale-service">
+          <div className="scale-service__units">
+            <input type="text" name="num-units" onChange={this._updateState}/>
```

Comment:
Can't change this to a number because then it puts the increment arrows on the right side of the inputs and modifies a number of other styles. We can revisit this later.